### PR TITLE
Live pipeline issues

### PR DESCRIPTION
**Pull request by seankane-msft**
Does a couple things here:
* regenerates code with newest version of Autorest
* regenerated code now passes the `set_table_access_policy` issue seen on the pipeline
* fixed up tests for new int32 serialization
* re-recordings because of the two above

### CODE REVIEW

**Review by kristapratico — APPROVED** [review]

File: sdk/tables/azure-data-tables/tests/test_table_async.py [hunk 1]
```diff
@@ -169,8 +169,6 @@ async def test_list_tables(self, resource_group, location, storage_account, stor
         # Assert
         for table_item in tables:
             self.assertIsInstance(table_item, TableItem)
-            self.assertIsNotNone(table_item.date)
```
Comment (iscai-msft) [inline sdk/tables/azure-data-tables/tests/test_table_async.py:172]: why was this removed

File: sdk/tables/azure-data-tables/tests/test_table_async.py [hunk 2]
```diff
@@ -242,7 +240,7 @@ async def test_list_tables_with_num_results(self, resource_group, location, stor
         async for s in ts.list_tables(results_per_page=3).by_page():
             small_page.append(s)
 
-        self.assertEqual(len(small_page), 2)
```
Comment (iscai-msft) [inline sdk/tables/azure-data-tables/tests/test_table_async.py]: why did this value change

File: sdk/tables/azure-data-tables/tests/test_table_entity.py [hunk 3]
```diff
@@ -148,15 +148,15 @@ def _assert_default_entity(self, entity, headers=None):
         '''
         Asserts that the entity passed in matches the default entity.
         '''
-        self.assertEqual(entity['age'], 39)
+        self.assertEqual(entity['age'].value, 39)
```
[the remark this answers is not in the document]
Comment (seankane-msft) [reply sdk/tables/azure-data-tables/tests/test_table_entity.py:151]: With the integer serialization changes are now wrapped in an `EntityProperty` class

File: sdk/tables/azure-data-tables/tests/test_table_entity_async.py [hunk 4]
```diff
@@ -1723,7 +1723,7 @@ async def test_sas_signed_identifier(self, resource_group, location, storage_acc
                 self.account_url(storage_account, "table"),
                 credential=token,
             )
-            table = service.get_table_client(table=self.table_name)
+            table = service.get_table_client(table_name=self.table_name)
```
[the remark this answers is not in the document]
Comment (seankane-msft) [reply sdk/tables/azure-data-tables/tests/test_table_entity_async.py:1726]: No, I'm not sure how this got through, the parameter has been called `table_name` for at least a month and this has just slipped under the radar. Probably because the live pipeline hasn't been up and running

File: sdk/tables/azure-data-tables/tests/test_table_async.py (hunk 2 above)
Comment (seankane-msft) [reply sdk/tables/azure-data-tables/tests/test_table_async.py]: This test is fickle and i'm still working through it. Essentially here there are two pages and each page is appended to the list. I do not know how this one slipped through either, probably because they were running from playback instead of live.
